outbox-v2/src/main/kotlin/com/ritense/outbox/autoconfigure/OutboxAutoConfiguration.kt

```diff
@@ -63,26 +66,48 @@ class OutboxAutoConfiguration {
     }
 
     @Bean
-    @ConditionalOnMissingBean(OutboxService::class)
-    fun outboxService(
+    @ConditionalOnMissingBean(DefaultOutboxService::class)
+    fun defaultOutboxService(
         outboxMessageRepository: OutboxMessageRepository,
         objectMapper: ObjectMapper,
-    ): OutboxService {
-        return OutboxService(
+    ): DefaultOutboxService {
+        return DefaultOutboxService(
             outboxMessageRepository,
             objectMapper
         )
     }
 
+    @Bean
+    @ConditionalOnMissingBean(CloudEventOutboxService::class)
+    fun cloudEventOutboxService(
+        defaultOutboxService: DefaultOutboxService
+    ): CloudEventOutboxService {
+        return CloudEventOutboxService(
+            defaultOutboxService
+        )
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(DomainEventOutboxService::class)
+    fun domainEventOutboxService(
+        cloudEventOutboxService: CloudEventOutboxService,
+        objectMapper: ObjectMapper
+    ): DomainEventOutboxService {
+        return DomainEventOutboxService(
+            cloudEventOutboxService,
+            objectMapper
+        )
+    }
+
     @Bean
     @ConditionalOnMissingBean(PollingPublisherService::class)
     fun pollingPublisherService(
-        outboxService: OutboxService,
+        defaultOutboxService: DefaultOutboxService,
```

Comment:
Why use the concrete class and not the interface. No there is no way of changing the 'one' outbox service